record/record_test.go

```
@@ -312,3 +313,137 @@ func TestStaleWriter(t *testing.T) {
 		t.Fatalf("stale write #1: unexpected error: %v", err)
 	}
 }
+
+func TestBasicReadRecovery(t *testing.T) {
+	buf := new(bytes.Buffer)
+	w := NewWriter(buf)
+
+	for i := 0; i < 3; i++ {
+		if wRec, err := w.Next(); err != nil {
+			t.Fatal(err)
+		} else {
+			if _, err = wRec.Write([]byte(strings.Repeat(string(rune(97+i)), blockSize-headerSize))); err != nil {
+				t.Fatal(err)
+			}
+		}
+	}
+	w.Close() // Finalize the last record.
+
+	rawBufSlice := buf.Bytes()
+
+	// Corrupt the checksum of the second record in our file.
+	rawBufSlice[blockSize] = 0xEF
+	rawBufSlice[blockSize+1] = 0xBE
+	rawBufSlice[blockSize+2] = 0xAD
+	rawBufSlice[blockSize+3] = 0xDE
+
+	// The first record should be read/processed just fine.
+	underlyingReader := bytes.NewReader(rawBufSlice)
+	r := NewReader(underlyingReader)
+	r1, err := r.Next()
+	if err != nil {
+		t.Fatal(err)
+	}
+	r1Data, err := ioutil.ReadAll(r1)
```

Don't ignore the error returned.